src/main.rs

```diff
@@ -101,6 +108,22 @@ fn work() -> Result<(), Cue2CCDError> {
 
     let tracks = cd.tracks();
 
+    let mut current_protection: Option<DiscProtection> = None;
+    let input = args.chosen_protection_type;
+
+    if args.protection {
+        match input {
```

Comment:
This can be simplified a little bit - Rust lets you match both if it's `Some` and the value of the string at the same time. For example:

```rust
match input {
  Some("discguard") => Some(DiscProtection::DiscGuard),
  // and so on
}
```

You can also assign and match in one action, so we could do something like:

```rust
let current_protection = match args.chosen_protection_type {
  Some("discguard") => Some(DiscProtection::DiscGuard),
  // and so on
  None => None,
};
```